Source/WebCore/Modules/mediastream/MediaStreamTrack.cpp

```diff
@@ -65,23 +65,15 @@
 
 namespace WebCore {
 
-static HashSet<MediaStreamTrack*>& allCaptureTracks()
-{
-    static NeverDestroyed<HashSet<MediaStreamTrack*>> captureTracks;
-    return captureTracks;
-}
-
-static MediaProducerMediaStateFlags sourceCaptureState(RealtimeMediaSource&);
-
 WTF_MAKE_ISO_ALLOCATED_IMPL(MediaStreamTrack);
 
 Ref<MediaStreamTrack> MediaStreamTrack::create(ScriptExecutionContext& context, Ref<MediaStreamTrackPrivate>&& privateTrack)
 {
     auto track = adoptRef(*new MediaStreamTrack(context, WTFMove(privateTrack)));
     track->suspendIfNeeded();
 
-    if (track->isCaptureTrack())
-        track->updateToPageMutedState();
+    if (track->isCaptureTrack() && !track->ended())
```

Comment:
```
if (RefPtr document = dynamicDowncast<Document>(context); track->isCaptureTrack() && !track->ended() && document)
  document->addCaptureSource...
```

would be nicer I believe

Reply:
A track can be a capture track only if its context is a Document, no need for a check here.